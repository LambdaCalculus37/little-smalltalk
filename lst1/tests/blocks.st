Class Main
[
	main
		(2 < 3) ifTrue: ['correct-1' print].
		((2 < 3) ifTrue: ['correct-2']) print.
		[:x | x print] value: 'correct-3' .
		((2 < 3) or: [3 < 4]) ifTrue: ['correct-4' print].
		((2 > 3) or: [3 < 4]) ifTrue: ['correct-5' print].
		((2 < 3) and: [3 < 4]) ifTrue: ['correct-6' print].
		((2 > 3) and: [3 < 4]) ifFalse: ['correct-7' print].
		self test1 print
|
	test1
		self test2: [^ 'correct-8'].
		'should not print' print
|
	test2: aBlock
		self test3: aBlock.
		'should not print' print
|
	test3: bBlock
		bBlock value.
		'should not print' print
]
