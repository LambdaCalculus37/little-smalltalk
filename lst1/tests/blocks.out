Little Smalltalk
	correct-1
correct-2
correct-3
correct-4
correct-5
correct-6
correct-7
correct-8
Main
	

